### PR TITLE
Added redis caching and minor changes to documentation

### PR DESCRIPTION
**Pull request by mijaz18**
Added redis caching and minor changes to the documentation with reference to this change

### CODE REVIEW

**Comment by mijaz18** [comment]
@stevedlawrence Apologies for last PR getting removed because I had to fork the repo again (My git log had become a mess and starting over seemed like a better option) Regarding your comments from the last PR:

1) Yes I agree the two example repos are very similar for now. In future, I will be adding deployment to Google Cloud App Engine via Terraform on the bigtable example. Our plan is to show different forms of deployment along with different cloud databases used (which is the case currently) in these examples! In that case, I guess it would make sense to keep the examples in different directories.

2) I have addressed the missing documentation for redis caching and missing new files names in the README in this PR

Thanks for all your suggestions, they are highly appreciated! Please let me know if have any more questions/concerns.

**Review by stevedlawrence — APPROVED** [review]
+1, just one small suggested change in the README

File: pubsub_bigtable_binary_processor_with_debugger/README.md [hunk 1]
```diff
@@ -38,17 +45,25 @@ Before you start is recommended that you install the following tools:
 2. [Cloud Bigtable Tool](https://cloud.google.com/bigtable/docs/cbt-overview)
 
 ## Technology Stack
+
 1. Cloud Bigtable
 2. Cloud Pubsub
+3.
+Redis/[Memorystore](https://codelabs.developers.google.com/codelabs/cloud-spring-cache-memorystore/#0)
```
[the remark this answers is not in the document]
Comment (stevedlawrence) [reply pubsub_bigtable_binary_processor_with_debugger/README.md]: This was a small enough fix, I just made fixed it myself.

**Comment by stevedlawrence** [comment]
Merged. Thanks!

**Pull request merged**